### PR TITLE
LUN-2744 Option to customize image select widget.

### PR DESCRIPTION
**Pull request by virtosubogdan**

### CODE REVIEW

File: filer/static/filer/js/popup_handling.js [hunk 1]
```diff
@@ -3,16 +3,31 @@
...
         var img_name = name + '_thumbnail_img';
         var txt_name = name + '_description_txt';
         var clear_name = name + '_clear';
+	var link_name = name + '_link_to_file';
         var elem = document.getElementById(name);
         document.getElementById(name).value = chosenId;
-        document.getElementById(img_name).src = chosenThumbnailUrl;
-        document.getElementById(txt_name).innerHTML = chosenDescriptionTxt;
+        img_element = document.getElementById(img_name);
+	use_full_image = img_element.attributes["data-use-full-image"];
```
Comment (alexandrupirjol) [inline filer/static/filer/js/popup_handling.js]: The indentation seems to be missing a space. See other lines around this one.
Also `img_element` and `use_full_image` are global if thay are not declared with `var`.  See also `description` and `link` bellow

File: filer/templates/admin/filer/widgets/admin_file_custom.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	django.jQuery("#{{ id }}").hide();
+	django.jQuery("#{{ clear_id }}").click(function(){
+		django.jQuery("#{{ id }}").removeAttr("value");
+		django.jQuery("#{{ thumb_id }}").attr("src", "{{ filer_static_prefix }}icons/nofile_48x48.png");
+		django.jQuery("#{{ span_id }}").html("");
+		django.jQuery("#{{ clear_id }}").hide();
+	});
+	django.jQuery(document).ready(function(){
+		var plus = django.jQuery("#add_{{ id }}");
+		if (plus.length){
```
Comment (alexandrupirjol) [inline filer/templates/admin/filer/widgets/admin_file_custom.html]: No need to check if jQuery element exists. jQuery handles this by default.

File: filer/static/filer/js/popup_handling.js [hunk 3]
```diff
@@ -3,16 +3,31 @@
...
+	if (use_full_image === undefined || !file_url) {
```
Comment (alexandrupirjol) [inline filer/static/filer/js/popup_handling.js]: `!use_full_image || !file_url` is more concise

File: filer/templates/admin/filer/folder/file_row.html [hunk 4]
```diff
@@ -2,7 +2,7 @@
 <td class="center">
 
     {% if is_popup and not select_folder %}
-        <a class="insertlink insertlinkButton" href="" onclick="opener.dismissRelatedImageLookupPopup(window, {{ file.id }}, '{{ file.icons.32|escapejs }}', '{{ file.label|escapejs }}'); return false;" title="{% trans "Select this file" %}"><i class="ace-icon fa fa-long-arrow-up bigger-125 blue"></i></a>
+        <a class="insertlink insertlinkButton" href="" onclick="opener.dismissRelatedImageLookupPopup(window, {{ file.id }}, '{{ file.icons.32|escapejs }}', '{{ file.label|escapejs }}', '{{ file.url }}'); return false;" title="{% trans "Select this file" %}"><i class="ace-icon fa fa-long-arrow-up bigger-125 blue"></i></a>
```
Comment (earthboundkid) [inline filer/templates/admin/filer/folder/file_row.html]: `file.url` is unlikely to have quotes in it, but it wouldn't hurt to use `|escapejs` as a precaution.

File: filer/static/filer/js/popup_handling.js (hunk 1 above)
Comment (virtosubogdan) [inline filer/static/filer/js/popup_handling.js]: Thanks.

File: filer/templates/admin/filer/folder/file_row.html (hunk 4 above)
Comment (virtosubogdan) [inline filer/templates/admin/filer/folder/file_row.html]: Good idea. Thanks.

**Pull request merged**